### ISSUE TITLE
Postgres chokes on specific external grading results

### ISSUE THREAD

**Issue by nwalters512** [PrairieLearn/PrairieLearn#931]
Filing so I remember to investigate later. Not sure if this is on PrairieLearn or the CS 225 grader to handle properly. Looks like there might be a null character in the JSON?

```
Nov 30 16:58:27 ip-10-0-0-204.us-east-2.compute.internal node[6879]: 2017-11-30T16:58:27.671Z - info: Submitting external grading job 4052803.
Nov 30 16:58:27 ip-10-0-0-204.us-east-2.compute.internal node[6879]: 2017-11-30T16:58:27.705Z - info: Successfully set up /jobs/job_4052803
Nov 30 16:58:32 ip-10-0-0-204.us-east-2.compute.internal node[6879]: 2017-11-30T16:58:32.055Z - error: processGradingResult: error message=unsupported Unicode escape sequence, stack=Async Stacktrace:
Nov 30 16:58:32 ip-10-0-0-204.us-east-2.compute.internal node[6879]: at async.series (/PrairieLearn/lib/assessment.js:356:17)
Nov 30 16:58:32 ip-10-0-0-204.us-east-2.compute.internal node[6879]: at sqldb.call (/PrairieLearn/lib/assessment.js:351:25)
Nov 30 16:58:32 ip-10-0-0-204.us-east-2.compute.internal node[6879]: at /PrairieLearn/lib/sqldb.js:271:17
Nov 30 16:58:32 ip-10-0-0-204.us-east-2.compute.internal node[6879]: at handleError (/PrairieLearn/lib/sqldb.js:220:17)
Nov 30 16:58:32 ip-10-0-0-204.us-east-2.compute.internal node[6879]: error: unsupported Unicode escape sequence
Nov 30 16:58:32 ip-10-0-0-204.us-east-2.compute.internal node[6879]: at Connection.parseE (/PrairieLearn/node_modules/pg/lib/connection.js:569:11)
Nov 30 16:58:32 ip-10-0-0-204.us-east-2.compute.internal node[6879]: at Connection.parseMessage (/PrairieLearn/node_modules/pg/lib/connection.js:396:17)
Nov 30 16:58:32 ip-10-0-0-204.us-east-2.compute.internal node[6879]: at Socket.<anonymous> (/PrairieLearn/node_modules/pg/lib/connection.js:132:22)
Nov 30 16:58:32 ip-10-0-0-204.us-east-2.compute.internal node[6879]: at emitOne (events.js:96:13)
Nov 30 16:58:32 ip-10-0-0-204.us-east-2.compute.internal node[6879]: at Socket.emit (events.js:191:7)
Nov 30 16:58:32 ip-10-0-0-204.us-east-2.compute.internal node[6879]: at readableAddChunk (_stream_readable.js:178:18)
Nov 30 16:58:32 ip-10-0-0-204.us-east-2.compute.internal node[6879]: at Socket.Readable.push (_stream_readable.js:136:10)
Nov 30 16:58:32 ip-10-0-0-204.us-east-2.compute.internal node[6879]: at TCP.onread (net.js:561:20), data={"sqlError":{"name":"error","length":206,"severity":"ERROR","code":"22P05","detail":"\\u0000 cannot be converted to text.","where":"JSON data, line 1: ...esults\":{\"score\":0,\"tests\":[{\"points\":0,\"output\":...","file":"json.c","line":"879","routine":"json_lex_string","message":"unsupported Unicode escape sequence"},"sql":"SELECT * FROM grading_jobs_process_external($1,$2,$3,$4,$5,$6)","sqlParams":[4052803,0,{"start_time":"2017-11-30T16:58:28.938Z","end_time":"2017-11-30T16:58:31.973Z","succeeded":true,"results":{"score":0,"tests":[{"points":0,"output":"0 number of times through\n1 number of times through\n2 number of times through\n3 number of times through\n4 number of times through\n5 number of times through\n6 number of times through\n7 number of times through\n8 number of times through\n9 number of times through\n10 number of times through\n11 number of times through\n12 number of times through\n13 number of times through\n14 number of times through\n15 number of times through\n16 number of times through\n17 number of times through\n18 number of times through\n19 number of times through\n20 number of times through\n21 number of times through\n22 number of times through\n23 number of times through\n24 number of times through\n25 number of times through\n26 number of times through\n27 number of times through\n28 number of times through\n29 number of times through\n30 number of times through\n31 number of times through\n32 number of times through\n33 number of times through\n34 number of times through\n35 number of times through\n36 number of times through\n37 number of times through\n38 number of times through\n39 number of times through\n40 number of times through\n41 number of times through\n42 number of times through\n43 number of times through\n44 number of times through\n45 number of times through\n46 number of times through\n47 number of times through\n48 number of times through\n49 number of times through\n50 number of times through\n51 number of times throug
Nov 30 16:58:32 ip-10-0-0-204.us-east-2.compute.internal node[6879]: h\n52 number of times through\n53 number of times through\n54 number of times through\n55 number of times through\n56 number of times through\n57 number of times through\n58 number of times through\n59 number of times through\n60 number of times through\n61 number of times through\n62 number of times through\n63 number of times through\n64 number of times through\n65 number of times through\n66 number of times through\n67 number of times through\n68 number of times through\n69 number of times through\n70 number of times through\n71 number of times through\n72 number of times through\n73 number of times through\n74 number of times through\n75 number of times through\n76 number of times through\n77 number of times through\n78 number of times through\n79 number of times through\n0 number of times through\n1 number of times through\n2 number of times through\n3 number of times through\n4 number of times through\n5 number of times through\n6 number of times through\n7 number of times through\n8 number of times through\n9 number of times through\n10 number of times through\n11 number of times through\n12 number of times through\n13 number of times through\n14 number of times through\n15 number of times through\n16 number of times through\n17 number of times through\n18 number of times through\n19 number of times through\n20 number of times through\n21 number of times through\n22 number of times through\n23 number of times through\n24 number of times through\n25 number of times through\n26 number of times through\n27 number of times through\n28 number of times through\n29 number of times through\n30 number of times through\n31 number of times through\n32 number of times through\n33 number of times through\n34 number of times through\n35 number of times through\n36 number of times through\n37 number of times through\n38 number of times through\n39 number of times through\n40 number of times through\n41 number of times through\n42 number of times through\n43 number of times through\n44 number of times through\n45 number o
Nov 30 16:58:32 ip-10-0-0-204.us-east-2.compute.internal node[6879]: f times through\n46 number of times through\n47 number of times through\n48 number of times through\n49 number of times through\n50 number of times through\n51 number of times through\n52 number of times through\n53 number of times through\n54 number of times through\n55 number of times through\n56 number of times through\n57 number of times through\n58 number of times through\n59 number of times through\n60 number of times through\n61 number of times through\n62 number of times through\n63 number of times through\n64 number of times through\n65 number of times through\n66 number of times through\n67 number of times through\n68 number of times through\n69 number of times through\n70 number of times through\n71 number of times through\n72 number of times through\n73 number of times through\n74 number of times through\n75 number of times through\n76 number of times through\n77 number of times through\n78 number of times through\n79 number of times through\n\n~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\ntests is a Catch v1.9.1 host application.\nRun with -? for options\n\n-------------------------------------------------------------------------------\nFifty random tests of increasing complexity.\n-------------------------------------------------------------------------------\ntests.cpp:26\n...............................................................................\n\ntests.cpp:31: FAILED:\n  REQUIRE( beiju(nu) == s_beiju(nu) )\nwith expansion:\n  \"pkmcoqhnwnkuewhsqmgbbuqcljjivswmdkqtbxixmvtrrbljtnsnfwzqfjmafadrrwsofsbcnuv-\n  qhff\u0000\"\n  ==\n  \"pkmcoqhnwnkuewhsqmgbbuqcljjivswmdkqtbxixmvtrrbljtnsnfwzqfjmafadrrwsofsbcnuv-\n  qhff\"\nwith message:\n  Input string was pkmcoqhnwnkuewhsqmgbbuqcljjivswmdkqtbxixmvtrrblj]\n  tnsnfwzqfjmafadrrwsofsbcnuvqhff\n\n===============================================================================\ntest cases: 1 | 1 failed\nassertions: 2 | 1 passed | 1 failed\n\n","name":"Fifty random tests of increasing complexity.","max_points":1}],"succ
Nov 30 16:58:32 ip-10-0-0-204.us-east-2.compute.internal node[6879]: eeded":true},"job_id":4052803,"received_time":"2017-11-30T16:58:28.328Z"},"2017-11-30T16:58:28.328Z","2017-11-30T16:58:28.938Z","2017-11-30T16:58:31.973Z"]}
```

**Comment by nwalters512** [comment]
(xposting from Slack)

It looks like postgres doesn’t allow null bytes in text fields, which I would imagine extends to json types as well: https://stackoverflow.com/questions/28813409/are-null-bytes-allowed-in-unicode-strings-in-postgresql-via-python. The best way to handle this might be to just strip all null bytes from course results?

**Comment by nwalters512** [comment]
Resolved by #935
